src/test/java/org/xbill/DNS/ZoneTest.java

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-2-Clause
+package org.xbill.DNS;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ZoneTest {
+  private Zone zone;
+
+  @BeforeEach
+  void beforeEach() throws IOException {
+    Name nameZone = new Name("example.");
+    InetAddress localhost4 = InetAddress.getByName("127.0.0.1");
+    InetAddress localhost6a = InetAddress.getByName("::1");
+    InetAddress localhost6b = InetAddress.getByName("::2");
+    Record[] zoneRecords =
+        new Record[] {
+          new SOARecord(
+              nameZone,
+              DClass.IN,
+              3600L,
+              Name.fromConstantString("nameserver."),
+              new Name("hostmaster", nameZone),
+              1,
+              21600L,
+              7200L,
+              2160000L,
+              3600L),
+          new NSRecord(nameZone, DClass.IN, 300L, Name.fromConstantString("nameserver.")),
+          new ARecord(new Name("test", nameZone), DClass.IN, 3600, localhost4),
+          new AAAARecord(new Name("test", nameZone), DClass.IN, 3600, localhost6a),
+          new AAAARecord(new Name("test", nameZone), DClass.IN, 3600, localhost6b),
+          new ARecord(new Name("*", nameZone), DClass.IN, 3600, localhost4),
+          new TXTRecord(new Name("*", nameZone), DClass.IN, 3600, "sometext")
+        };
+    zone = new Zone(nameZone, zoneRecords);
+  }
+
+  @Test
+  void exactNameExistingALookup() {
+    Name testName = Name.fromConstantString("test.example.");
+    SetResponse resp = zone.findRecords(testName, Type.A);
+    assertEquals(testName, resp.answers().get(0).first().getName());
```

Comment:
In general I think that leaning on the the excellent list comparison abilities of `assertEquals` is preferable to looking at specific properties of the returned answer, but since `RRset` doesn't have a succinct way of creating an instance and is also missing equals and hash methods there would be some work to get that into the code.

In this case though, the tests are such a clear improvement so feel free to land.

Reply:
equals/hashCode of rrset is a one-liner, so I tried this now. It makes constructing the zone much more awkward with the intermediate construction of RRsets and the only test that really benefits is exactNameAnyLookup. I've created  #174 for comparison. Have a look please, I'm not sure what I like better.

Reply:
Oh interesting. I'll comment over there.